### ISSUE TITLE
amplify rebuild api - crashes

### ISSUE THREAD

**Issue by kevcle** [aws-amplify/amplify-cli#9151]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### How did you install the Amplify CLI?

npm install -g @aws-amplify/cli

### If applicable, what version of Node.js are you using?

v14.17.2

### Amplify CLI Version

7.5.4

### What operating system are you using?

Ubuntu under WSL2

### Amplify Categories

api

### Amplify Commands

Not applicable

### Describe the bug

Running the **amplify rebuild api** CLI command fails.  This command was documented in the documentation for v1 of the graphql transformer, but are not documented in the new version of the documentation.

```
UPDATE_COMPLETE GraphQLAPITransformerSchema3CB2AE18 AWS::AppSync::GraphQLSchema Tue Nov 30 2021 09:52:23 GMT-0800 (Pacific Standard Time) 
✖ An error occurred when pushing the resources to the cloud
🛑 ["Index: 0 State: {\"deploy\":\"waitingForDeployment\"} Message: Resource is not in the state stackUpdateComplete"]
DeploymentError: ["Index: 0 State: {\"deploy\":\"waitingForDeployment\"} Message: Resource is not in the state stackUpdateComplete"]
    at /home/kevcle/.nvm/versions/node/v14.17.2/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/src/iterative-deployment/deployment-manager.ts:159:40
    at Interpreter.update (/home/kevcle/.nvm/versions/node/v14.17.2/lib/node_modules/@aws-amplify/cli/node_modules/xstate/lib/interpreter.js:265:9)
    at /home/kevcle/.nvm/versions/node/v14.17.2/lib/node_modules/@aws-amplify/cli/node_modules/xstate/lib/interpreter.js:110:15
    at Scheduler.process (/home/kevcle/.nvm/versions/node/v14.17.2/lib/node_modules/@aws-amplify/cli/node_modules/xstate/lib/scheduler.js:69:7)
    at Scheduler.flushEvents (/home/kevcle/.nvm/versions/node/v14.17.2/lib/node_modules/@aws-amplify/cli/node_modules/xstate/lib/scheduler.js:60:12)
    at Scheduler.schedule (/home/kevcle/.nvm/versions/node/v14.17.2/lib/node_modules/@aws-amplify/cli/node_modules/xstate/lib/scheduler.js:49:10)
    at Interpreter.send (/home/kevcle/.nvm/versions/node/v14.17.2/lib/node_modules/@aws-amplify/cli/node_modules/xstate/lib/interpreter.js:104:23)
    at actor.id (/home/kevcle/.nvm/versions/node/v14.17.2/lib/node_modules/@aws-amplify/cli/node_modules/xstate/lib/interpreter.js:1006:15)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:95:
```

### Expected behavior

Redeploy an empty set of the DDB tables backing the graphQL API. as documented here: https://docs.amplify.aws/cli-legacy/graphql-transformer/overview/#update-schema

Functionally similar to removing and readding the API.

### Reproduction steps

1. create a new project with amplify init
2. amplify add api to create a new graphqlapi
3. amplify push to deploy the api
4. amplify rebuild api

This is a spurious example but repos the error.  A normal use case would include running mutations to manipulate the underlying data.

### GraphQL schema(s)

<details>

```graphql
# Put schemas below this line
# This "input" configures a global authorization rule to enable public access to
# all models in this schema. Learn more about authorization rules here: https://docs.amplify.aws/cli/graphql/authorization-rules
input AMPLIFY {
	globalAuthRule: AuthRule = { allow: public }
} # FOR TESTING ONLY!
enum IngestionStatus {
	pending
	active
	success
	failure
}

enum ItemType {
	multiplechoice
	multipleresponse
}

enum ExamType {
	practice
	highstakes
}

enum ExamFamily {
	foundational
	associate
	professional
	specialty
}

type ItemFile @model {
	itemLoadId: String! @primaryKey
	fileLocation: String! @index(name: "byFileLocation", queryField: "itemFilesByLocation")
	status: IngestionStatus
	statusMessage: String
	ingestedIterations: [ExamItemIteration] @hasMany(indexName: "bySourceFile", fields: ["itemLoadId"])
}

type ItemBank @model {
	bankId: String! @primaryKey
	bankName: String!
	bankDescription: String!
	bankType: ExamType!
	bankFamily: ExamFamily!
	bankItems: [ExamItem] @hasMany(indexName: "byBank", fields: ["bankId"])
}

type ExamItem @model {
	examItemId: String! @primaryKey
	currentIterationId: String!
	owningBankId: String! @index(name: "byBank", sortKeyFields: ["examItemId"])
	owningBank: ItemBank @belongsTo(fields: ["owningBankId"])
	itemIterations: [ExamItemIteration] @hasMany(indexName: "byItem", fields: ["examItemId"])
}

type ExamItemIteration @model {
	examItemIterationId: String! @primaryKey
	owningItemId: String! @index(name: "byItem", sortKeyFields: ["examItemIterationId"])
	owningItem: ExamItem @belongsTo(fields: ["owningItemId"])

	owningFileId: String! @index(name: "bySourceFile")
	owningFile: ItemFile @belongsTo(fields: ["owningFileId"])

	iterationNumber: Int!
	questionNumber: String!
	itemType: ItemType
	topic: String!
	stem: String!
	rationale: String
	keys: String
	minResponses: Int
	maxResponses: Int
	status: String!
	enemyItemList: String!
	responses: [ExamItemResponse] @hasMany(indexName: "byItemIteration", fields: ["examItemIterationId"])
}

type ExamItemResponse @model {
	examItemResponseId: String! @primaryKey
	owningItemIterationId: String! @index(name: "byItemIteration", sortKeyFields: ["sequence"])
	owningItemIteration: ExamItemIteration @belongsTo(fields: ["owningItemIterationId"])
	sequence: Int!
	responseKey: String!
	content: String
	rationale: String
	isCorrect: Boolean
}




```

</details>

### Log output

<details>

```
# Put your logs below this line

🛑 ["Index: 0 State: {\"deploy\":\"waitingForDeployment\"} Message: Resource is not in the state stackUpdateComplete"]
DeploymentError: ["Index: 0 State: {\"deploy\":\"waitingForDeployment\"} Message: Resource is not in the state stackUpdateComplete"]
    at /home/kevcle/.nvm/versions/node/v14.17.2/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/src/iterative-deployment/deployment-manager.ts:159:40
    at Interpreter.update (/home/kevcle/.nvm/versions/node/v14.17.2/lib/node_modules/@aws-amplify/cli/node_modules/xstate/lib/interpreter.js:265:9)
    at /home/kevcle/.nvm/versions/node/v14.17.2/lib/node_modules/@aws-amplify/cli/node_modules/xstate/lib/interpreter.js:110:15
    at Scheduler.process (/home/kevcle/.nvm/versions/node/v14.17.2/lib/node_modules/@aws-amplify/cli/node_modules/xstate/lib/scheduler.js:69:7)
    at Scheduler.flushEvents (/home/kevcle/.nvm/versions/node/v14.17.2/lib/node_modules/@aws-amplify/cli/node_modules/xstate/lib/scheduler.js:60:12)
    at Scheduler.schedule (/home/kevcle/.nvm/versions/node/v14.17.2/lib/node_modules/@aws-amplify/cli/node_modules/xstate/lib/scheduler.js:49:10)
    at Interpreter.send (/home/kevcle/.nvm/versions/node/v14.17.2/lib/node_modules/@aws-amplify/cli/node_modules/xstate/lib/interpreter.js:104:23)
    at actor.id (/home/kevcle/.nvm/versions/node/v14.17.2/lib/node_modules/@aws-amplify/cli/node_modules/xstate/lib/interpreter.js:1006:15)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
```

</details>

### Additional information

Workaround is :
amplify remove ap
amplify push
amplify add api (use the existing schema)
amplify push

**Comment by Diverious** [comment]
I get this error all of the time.  I keep having to recreate a completely new app.  Is there anyway to fix this?

**Comment by nallibtermatrac** [comment]
Getting the same error.
I pulled from cloud... added a new function.... impossible to push now.
Please, 1 week trying to launch my first project !

**Comment by lehoai** [comment]
any new?

**Comment by rafaelfaria** [comment]
I LITERALLY just started getting this error. Like... out of the blue.

**Comment by malaganator3000** [comment]
any update? :(

**Comment by daviddalmaso** [comment]
Same error... Anyone able to solve this?

**Comment by clintfoster** [comment]
Same here (v2 transformer). I had not made any changes to my schemas. I just wanted to wipe out my tables and start over.

**Comment by naingaungphyo** [comment]
I also got the same _"Resource is not in the state stackUpdateComplete"_ error after changing many indexes and the primary key of one of my models.
I tried with [this](https://docs.amplify.aws/cli/graphql/troubleshooting/#deploying-multiple-index-changes-at-once) amplify/cli/json setting as `true` and also used `--force` and `--allow-destructive-graphql-schema-updates` cli flags according to [this](https://docs.amplify.aws/cli/project/troubleshooting/#push-failures-from-unmanaged-changes-to-cloud-resources---drift) troubleshooting guide, but none of them works.

By the way, I didn't do any manual modification. (eg. from console etc)

**My workaround** was removing the api and adding api again as below.
1. backup my schema file
2. amplify remove api
3. amplify push
4. amplify add api and use my backup schema
5. amplify push

used cli version is `4.3.20`
